client/layout/masterbar/logged-in.jsx

```diff
@@ -94,6 +94,7 @@ class MasterbarLoggedIn extends React.Component {
 	clickMySites = () => {
 		this.props.recordTracksEvent( 'calypso_masterbar_my_sites_clicked' );
 		this.handleLayoutFocus( 'sites' );
+		this.props.activateNextLayoutFocus();
```

Comment:
This is because the activateNextLayoutFocus runs on route change in sections-middleware.js but in some cases there isn't a URL for the route to change to. See this in action [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/sections-middleware.js#L98)

Reply:
The URL is set here 
https://github.com/Automattic/wp-calypso/blob/f798ce57dbcf909766de015b2ed43d54aefee616/client/layout/masterbar/logged-in.jsx#L195-L197

it is set to `mySitesUrl = ''` as a workaround so that we avoid changing URL when for example you are in `/settings` and you try opening the sidebar (applicable only when in `my-sites` section). 

I am not sure whether this change is needed. Can you elaborate?

Reply:
Hmmm... this is still needed because in the Item component we are checking to see if the url is empty or not, in order to determine if we should display a button or a link. We could move it of course, but maybe we should keep the Item component decoupled of this condition? 🤔

Reply:
I am not sure we are referring to the same thing Bogdan?
`this.handleLayoutFocus( 'sites' );` is already triggering `setNextLayoutFocus()`. Why do we also need `activateNextLayoutFocus`?

Reply:
@cpapazoglou AFAIK there is a difference between between _setting_ the next layout focus and _activating_ the next layout focus.

By setting it I believe you are making sure that when it's next activated (which is what we want to do, to toggle the sidebar) that it's readily set the value in advance.

Usually the set value is activated on route change [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/sections-middleware.js#L26) but because there isn't _always_ going to be a route change (because we removed the empty anchor href - it was being pre-populated with an incorrect URL at times, see [here](https://github.com/Automattic/wp-calypso/pull/54203#discussion_r662193654)) we activated it in here.